### PR TITLE
Update Azure cluster-autoscaler e2e test

### PR DESCRIPTION
**Pull request by nojnhuh**
#### What type of PR is this?
/kind bug

#### What this PR does / why we need it:

This PR makes some changes to spruce up the output of the cluster-autoscaler Azure test.
- Generate junit report
- Remove debug log statement
- Wait for NotReady Nodes not to exist at the end of the test

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: cluster-autoscaler/cloudprovider/azure/test/e2e/e2e_suite_test.go [hunk 1]
```diff
@@ -77,14 +77,12 @@ func allVMSSStable(g Gomega) {
...
-			ready := false
 			for _, cond := range node.Status.Conditions {
 				if cond.Type == corev1.NodeReady && cond.Status == corev1.ConditionTrue {
-					ready = true
-					break
+					return true
 				}
 			}
-			return ready
+			return false
```
Comment (jackfrancis) [inline cluster-autoscaler/cloudprovider/azure/test/e2e/e2e_suite_test.go:85]: wow this is a neat trick

**Comment by jackfrancis** [comment]
/lgtm
/approve

feel free to unhold if/after test passes

**Comment by nojnhuh** [comment]
Looks good!
/hold cancel

**Pull request merged**